tests/unit/commands/samconfig/test_samconfig.py

```diff
@@ -0,0 +1,483 @@
+            "docker_network": "mynetwork",
+            "log_file": "logfile",
+            "layer_cache_basedir": "basedir",
+            "skip_pull_image": True,
+            "force_image_build": True,
+            "parameter_overrides": "ParameterKey=Key,ParameterValue=Value",
+        }
+
+        # NOTE: Because we don't load the full Click BaseCommand here, this is mounted as top-level command
+        with samconfig_parameters(["start-lambda"], self.scratch_dir, **config_values) as config_path:
+
+            from samcli.commands.local.start_lambda.cli import cli
+
+            LOG.debug(Path(config_path).read_text())
+            runner = CliRunner()
+            result = runner.invoke(cli, [])
+
+            LOG.info(result.output)
+            LOG.info(result.exception)
+            if result.exception:
+                LOG.exception("Command failed", exc_info=result.exc_info)
+            self.assertIsNone(result.exception)
+
+            do_cli_mock.assert_called_with(
+                ANY,
+                "127.0.0.1",
+                12345,
+                str(Path(os.getcwd(), "mytemplate.yaml")),
+                "envvar.json",
+                (1, 2, 3),
+                "args",
+                "mypath",
+                "basedir",
+                "mynetwork",
+                "logfile",
+                "basedir",
+                True,
+                True,
+                {"Key": "Value"},
+            )
+
+    @patch("samcli.commands.package.command.do_cli")
+    def test_package(self, do_cli_mock):
+
+        config_values = {
+            "template_file": "mytemplate.yaml",
+            "s3_bucket": "mybucket",
+            "force_upload": True,
+            "s3_prefix": "myprefix",
+            "kms_key_id": "mykms",
+            "use_json": True,
+            "metadata": '{"m1": "value1", "m2": "value2"}',
+            "region": "myregion",
+            "output_template_file": "output.yaml",
+        }
+
+        with samconfig_parameters(["package"], self.scratch_dir, **config_values) as config_path:
+
+            from samcli.commands.package.command import cli
+
+            LOG.debug(Path(config_path).read_text())
+            runner = CliRunner()
+            result = runner.invoke(cli, [])
+
+            LOG.info(result.output)
+            LOG.info(result.exception)
+            if result.exception:
+                LOG.exception("Command failed", exc_info=result.exc_info)
+            self.assertIsNone(result.exception)
+
+            do_cli_mock.assert_called_with(
+                str(Path(os.getcwd(), "mytemplate.yaml")),
+                "mybucket",
+                "myprefix",
+                "mykms",
+                "output.yaml",
+                True,
+                True,
+                {"m1": "value1", "m2": "value2"},
+                "myregion",
+                None,
+            )
+
+    @patch("samcli.commands.deploy.command.do_cli")
+    def test_deploy(self, do_cli_mock):
```

Comment:
can we add another series of tests, where we supply args and they override whats in the config file.